### PR TITLE
Fix multiple 64-bit dotnet issues

### PR DESCRIPTION
**Pull request by austin987**
This should greatly improve the .Net Experience on win64.

### CODE REVIEW

File: src/winetricks [hunk 1]
```diff
@@ -6682,7 +6685,9 @@ load_dotnet45()
     w_call remove_mono
 
     # See https://appdb.winehq.org/objectManager.php?sClass=version&iId=25478 for Focht's recipe
-    w_call dotnet35
+
+    # Seems unneeded in wine-2.0
```
Comment (qwertychouskie) [inline src/winetricks]: As far as I know, this is only fixed in staging, see https://bugs.winehq.org/show_bug.cgi?id=26757.

File: src/winetricks [hunk 2]
```diff
@@ -6736,7 +6741,9 @@ load_dotnet452()
...
```
Comment (qwertychouskie) [inline src/winetricks]: Same here.

File: src/winetricks (hunk 1 above)
Comment (qwertychouskie) [inline src/winetricks]: As far as I know, this is only fixed in staging, see https://bugs.winehq.org/show_bug.cgi?id=26757.

File: src/winetricks [hunk 3]
```diff
@@ -6622,7 +6622,10 @@ w_metadata dotnet40 dlls \
 
 load_dotnet40()
 {
-    w_package_unsupported_win64
```
Comment (qwertychouskie) [inline src/winetricks]: Maybe add a `w_package_warn_win64`?

File: src/winetricks (hunk 1 above)
Comment (austin987) [inline src/winetricks]: That bug refers to allowing .msu installers to work. That's unrelated to if .net45 itself can install.

I've never used staging on a regular basis, but I can tell you that wine-2.4-55-g5b4f8a3 + this version of winetricks allows dotnet45 to install (with a 32-bit prefix, previously also tested with win64).

File: src/winetricks (hunk 2 above)
Comment (austin987) [inline src/winetricks]: See above.

File: src/winetricks (hunk 3 above)
Comment (austin987) [inline src/winetricks]: Good idea.

**Comment by austin987** [comment]
@qwertychouskie I've added w_package_broken_win64 (broken on wine) and w_package_warn_win64() (may or may not work on win64), and used that for .Net.

I've also re-enabled dotnet46* in the tests, removing dotnet35 seems to have helped there.

I'd still like to see some users test this. I've pinged all related bugs, but no one has given any results yet. For as many people that complained about this, you'd think someone would test it ;)

**Comment by austin987** [comment]
This fixed #756

File: src/winetricks (hunk 1 above)
Comment (qwertychouskie) [inline src/winetricks]: From my understanding based on bug reports, etc. the issue is that the .NET installers, while using a Wine-compatible install method for that version, it uses MSUs to install the previous versions. In Windows, this means that installing .NET 4.5 installs .NET 4.5, 4.0, 3.5, 3.0, 2.0, and 1.1.  In Wine non-staging (or staging < 1.8), the MSU installer is just a stub that fakes a successful install, so the .NET installation seems to go fine, but is incomplete.  The installer file should have its contents extracted and the MSUs examined to make sure no files are needed from .NET 3.5 or below.

Comment (austin987) [inline src/winetricks]: --verify was used, which means the dotnet_verifier was run. That only checks the given .Net version though, not earlier versions.

**Comment by fxthomas** [comment]
Just tested this because I needed .NET to work on a new Wine prefix. So far I've had no issues!

**Comment by austin987** [comment]
Committed as https://github.com/Winetricks/winetricks/commit/ddf5c7c059ebc9b78031584cb82332e091c52d28

**Comment by Pirat90pl** [comment]
I feel so dumb, dont know how to run it. Teach me please!

**Pull request closed without merging**